### ISSUE TITLE
Find Hub by Name

### ISSUE THREAD

**Issue by tthiery** [sharpbrick/powered-up#59]
By BLE advertisment name

Lessons Learnt from #49

**Comment by tthiery** [comment]
BLE advertisment does not include a name field. The device actually has it. Query device during discovery

**Comment by dlech** [comment]
Actually, the advertisement data does include "Local Name" in the scan response data.

**Comment by tthiery** [comment]
Thanks for noticing. I also looked into it. The LocalName was empty for me. I will have a second look into it. Maybe I need to adjust the watcher or something else so the field gets populated.

**Comment by GianCann** [comment]
It use the ShortName.

**Comment by GianCann** [comment]
@tthiery  take a look here: https://github.com/tessel/bleadvertise

**Comment by dlech** [comment]
There are two different advertisements, one with the `ScanResponse` flag set and one without (https://docs.microsoft.com/en-us/uwp/api/windows.devices.bluetooth.advertisement.bluetoothleadvertisementtype?view=winrt-19041). I forget the terminology, but I think you have to do something like an "active" scan to get the scan response type data.

**Comment by tthiery** [comment]
After reading, testing etc ... I came to a surprising end ;)

So WinRT API is a bit ... strange.

- I was actually actively scanning for advertisments `watcher.ScanningMode = BluetoothLEScanningMode.Active;`.
- The advertisment `LocalName` would need to be in a data section with DataType 8 (shortened) or 9 (complete). However, the TechnicMediumHub only returns 1 (Flags), 7 (CompleteService128BitUuids) and 255 (ManufacturerSpecificData).
- Then I was trying to install a watcher for the device from the windows internal registry of devices (unrelated to BT itself). That was painful and I failed doing it. [I am not alone](https://github.com/microsoft/Windows-universal-samples/issues/1089) with that ;)
- Then I looked it from a different perspective: If Windows has a device (and it has it) representing my hub, why I cannot retrieve it by a method using the Bluetooth Address. This again lead to the following code ...
  ````csharp
  using (var device = BluetoothLEDevice.FromBluetoothAddressAsync(eventArgs.BluetoothAddress).AsTask().Result)
  {
      info.Name = device.Name;
  }
  ````
  ... which is literally what I did before the complete excurse. According to the [docs](https://docs.microsoft.com/en-us/uwp/api/windows.devices.bluetooth.bluetoothledevice.frombluetoothaddressasync?view=winrt-19041) this should not even create a connection.

I let this open for another day or two ... some more experiments ... but i may let it settle like that.
